src/main/java/jenkins/plugins/slack/SlackService.java

```diff
@@ -15,5 +15,7 @@
 
     boolean publish(String message, JSONArray attachments, String color, String timestamp);
 
+    boolean publish(String message, JSONArray attachments, JSONArray blocks, String color, String timestamp);
```

Comment:
what's wrong with using the `publish(SlackRequest slackRequest)` it's supposed to make it so that we don't need to add any more methods here?

I don't like attachments and blocks being in the same method as they can't both be used at the same time

Reply:
This initial refactor turned `publish(SlackRequest)` into a general "make an API request" call, which meant that certain `slackSend`-specific logic (like the `replyBroadcast` option) had to be moved out. Let me try walking that back a bit so that `publish(SlackRequest)` routes through a `makeApiRequest(JSONBody)`. That should make it possible for this PR to leave all the usage of `publish()` alone.